src/mono/mono/mini/aot-compiler.c

```diff
@@ -5154,6 +5176,37 @@ add_wrappers (MonoAotCompile *acfg)
 	}
 }
 
+static void
+add_managed_to_native_wrappers (MonoAotCompile *acfg)
+{
```

Comment:
Shouldn't add_wrappers () call this ?

Reply:
add_wrappers is only called from one location, and that is where we now also calls add_managed_to_native_wrappers as well as add_native_to_managed_wrappers. I initially moved all into add_wrappers and added condition in add wrappers to only call the wrappers under full aot, but that diff caused a lot of noise to the overall PR, but let me try that again and see if I can get the diff a little cleaner.